conflation/map_matching.py

```diff
@@ -0,0 +1,292 @@
+import multiprocessing
+import os
+import pickle
+import requests
+from conflation import util
+
+VALHALLA_MAP_MATCHING_URL_EXTENSION = "trace_attributes"
+
+MAXIMUM_UNMATCHED_PERCENTAGE = (
+    0.25  # If more than 25% of points are unmatched, skip this sequence
+)
+MAXIMUM_SPEED = 160  # km / h, to weed out poor measurements and trains
+DENSITY_CLASSIFICATIONS = [  # The different density classifications we can give to roads
+    "rural",
+    "suburban",
+    "urban",
+]
+
+
+def run(
+    traces_dir: str,
+    map_matches_dir: str,
+    processes: int,
+    valhalla_url: str,
+    valhalla_headers: dict,
+) -> None:
+    """
+    This method is the second step in the script. It takes the traces from the first step and performs "map matching",
+    which is the process of determining which real-life roads the GPS traces map to. We use the open source Valhalla
```

Comment:
nit: to me it would make sense to rename this script to `valhalla.py`, similar to how you have it as `mapillary.py`. maybe even they could be in subdirs of `map_matching` and `gps_fetching` respectively? that way others can add new providers for both stages over time